lib/schema_checker.py

```diff
@@ -0,0 +1,118 @@
+import os
+import string
+import re
+from schema import Schema, SchemaError, Optional, Or, Use
+# https://docs.green-coding.berlin/docs/measuring/usage-scenario/
+# networks documentation is different than what i see in the wild!
+    # name: str
+    # also isn't networks optional?
+    # fix documentation - name not needed, also netowrks optional
+    # add check in runner.py networks parsing, make sure its valid_string
+    # is services/type optional?
+
+# services /type missing from documentation?
+
+
+# https://github.com/compose-spec/compose-spec/blob/master/spec.md
+
+def single_or_list(value):
+    return Or(value, [value])
+
+def is_valid_string(value):
+    valid_chars = set(string.ascii_letters + string.digits + '_' + '-')
+    if not set(value).issubset(valid_chars):
+        raise SchemaError(f"{value} does not use valid characters! (a-zA-Z0-9_-)")
+
+def contains_no_invalid_chars(value):
+    bad_values = re.findall(r'(\.\.|\$|\'|"|`|!)', value)
+    if bad_values:
+        raise SchemaError(f"{value} includes disallowed values: {bad_values}")
+
+## If we ever want smarter validation for ports, here's a list of examples that we need to support:
+# - 3000
+# - 3000-3005
+# - 8000:8000
+# - 9090-9091:8080-8081
+# - 49100:22
+# - 8000-9000:80
+# - 127.0.0.1:8001:8001
+# - 127.0.0.1:5000-5010:5000-5010
+# - 6060:6060/udp
+
+def validate_compose_include(value):
+    #!include compose.yml
+    # make sure value starts with the string '!include', then space, then a valid yaml file name
+    # make sure that compose yaml exists
+    if not value.startswith('!include '):
+        raise SchemaError(f"{value} does not begin with !include")
+    if (not value.endswith('.yml') and not value.endswith('.yaml')):
+        raise SchemaError(f"{value} does not end with a valid yaml filename")
+
+    # Extract the filename from the value
+    filename = value[len('!include '):]
+
+    # Make sure that the compose yaml exists
+    if not os.path.exists(filename):
+        raise SchemaError(f"{filename} does not exist")
+    return value
+
+def valid_service_types(value):
+    if value != 'container':
+        raise SchemaError(f"{value} is not 'container'")
+    return value
+
+usage_scenario_schema = Schema({
+    Optional("name"): str,
```

Comment:
I think we need description here too. Are name etc optional? Or is this just the way Schema does this?

Reply:
Added a comment - but yes, everything that is Optional() is not needed - took this from our docs requirements. Lemme know if any of these are actually required

Reply:
I think they are actually required. Or they are in the code. [1] Could you please update the documentation accordingly. 

[1] https://github.com/green-coding-berlin/green-metrics-tool/blob/main/runner.py#L299

Reply:
Made these values mandatory now